agent/api/ecsclient/client.go

```diff
@@ -230,6 +236,34 @@ func (client *APIECSClient) registerContainerInstance(clusterRef string, contain
 	return aws.StringValue(resp.ContainerInstance.ContainerInstanceArn), err
 }
 
+// disableTaskENIEnabledIfNoInit disables the Task ENI capability of the
+// Agent if the Agent was not started with an init system. We need the init system
+// to properly cleanup the dhclient processes when Task ENI capability is
+// enabled
+func (client *APIECSClient) disableTaskENIEnabledIfNoInit() {
+	ok, err := isAgentAlsoTheInit()
+	if err != nil {
+		seelog.Infof("Unable to determine if Agent is started with an init system; Disabling Task ENI capability: %v", err)
```

Comment:
I think this should be either WARN or ERROR.  This is actually a failure of being able to detect.